pkg/engine/imageVerify.go

```diff
@@ -66,21 +100,23 @@ func VerifyAndPatchImages(policyContext *PolicyContext) (*response.EngineRespons
 
 		logger.V(3).Info("processing image verification rule", "ruleSelector", applyRules)
 
+		var err error
+		ruleImages, err := requireValidation(policyContext, rule)
+		if err != nil {
+			appendError(resp, rule, fmt.Sprintf("failed to extract images: %s", err.Error()), response.RuleStatusError)
+			continue
+		}
+		if len(ruleImages) == 0 {
+			appendError(resp, rule, "skip run verification for image", response.RuleStatusSkip)
+			continue
+		}
+
 		policyContext.JSONContext.Restore()
 		if err := LoadContext(logger, rule.Context, policyContext, rule.Name); err != nil {
 			appendError(resp, rule, fmt.Sprintf("failed to load context: %s", err.Error()), response.RuleStatusError)
 			continue
 		}
 
-		ruleImages := images
-		var err error
-		if rule.ImageExtractors != nil {
-			if ruleImages, err = policyContext.JSONContext.GenerateCustomImageInfo(&policyContext.NewResource, rule.ImageExtractors); err != nil {
-				appendError(resp, rule, fmt.Sprintf("failed to extract images: %s", err.Error()), response.RuleStatusError)
-				continue
-			}
-		}
-
 		if ruleImages == nil {
```

Comment:
doesn't line 109 `if len(ruleImages) == 0 {` already perform this check?

Reply:
removed check